R/register_extendr.R

```diff
@@ -40,27 +45,37 @@ register_extendr <- function(path = ".", quiet = FALSE, force_wrappers = FALSE,
 
   outfile <- rprojroot::find_package_root_file("R", "extendr-wrappers.R", path = path)
 
-  # If force_wrappers is TRUE, use a minimal wrapper file even when
-  # make_wrappers() fails; since the wrapper generation depends on the compiled
-  # Rust code, the package needs to be installed before attempting this, but
-  # it's not always the case (e.g. the package might be corrupted, or not
-  # installed yet).
-  if (isTRUE(force_wrappers)) {
-    error_handle <- function(e) {
-      cli::cli_alert_danger("Failed to generate wrapper functions: {e$message}.")
-      cli::cli_alert_warning("Falling back to a minimal wrapper file instead.")
-
-      make_example_wrappers(pkg_name, outfile, path = path)
-    }
-  } else {
-    error_handle <- function(e) {
-      ui_throw(
-        "Failed to generate wrapper functions.",
-        c(
-          ui_x(e[["message"]])
-        )
-      )
-    }
+  path <- rprojroot::find_package_root_file(path = path)
+
+  # If compile is NA, compile if the DLL is newer than the source files
+  if (isTRUE(is.na(compile))) {
+    compile <- needs_compilation(path, quiet) || pkgbuild::needs_compile(path)
+  }
+
+  if (isTRUE(compile)) {
+    # This relies on [`pkgbuild::needs_compile()`], which
+    # does not know about Rust files modifications.
+    # `force = TRUE` enforces compilation.
+    pkgbuild::compile_dll(
+      path = path,
+      force = TRUE,
+      quiet = quiet
+    )
+  }
+
+  dll_file <- get_library_path(path)
+
+  # If the wrapper file is newer than the DLL file, assume it's generated by the
```

Comment:
Write "assume it's been generated" to make the chain of causality absolutely clear?